src/modules/appointment/appointment.service.ts

```diff
@@ -0,0 +1,143 @@
+import { HttpException, HttpStatus, Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository, SelectQueryBuilder } from 'typeorm';
+import DoctorService from 'modules/doctor/doctor.service';
+import PatientService from 'modules/patient/patient.service';
+import Patient from 'modules/patient/entity/patient.entity';
+import Appointment from './entity/appointment.entity';
+import CreateAppointmentDto from './dto/create-appointment.dto';
+
+@Injectable()
+export default class AppointmentService {
+  constructor(
+    @InjectRepository(Appointment)
+    private appointmentRepository: Repository<Appointment>,
+    private readonly doctorService: DoctorService,
+    private readonly patientService: PatientService,
+  ) {}
+
+  async createAppointment(
+    appointment: CreateAppointmentDto,
+  ): Promise<Appointment> {
+    try {
+      const { localDoctorId, remoteDoctorId, patientId, ...rest } = appointment;
+
+      const patient = await this.patientService.getPatientById(patientId);
+      const localDoctor = await this.doctorService.getDoctorByID(localDoctorId);
+      const remoteDoctor = await this.doctorService.getDoctorByID(
+        remoteDoctorId,
+      );
+
+      const newAppointment = {
+        localDoctor,
+        remoteDoctor,
+        patient,
+        ...rest,
+      };
+
+      return await this.appointmentRepository
+        .createQueryBuilder()
+        .insert()
+        .into(Appointment)
+        .values(newAppointment)
+        .execute()
+        .then((result) => {
+          const { generatedMaps } = result;
+          const [generatedMap] = generatedMaps;
+          const { id } = generatedMap;
+
+          return { ...newAppointment, id };
+        });
```

Comment:
find other way, make 3 select query to make one insert too expencive

Reply:
rewrite with single query